lib/core/src/model.rs

```diff
@@ -951,19 +977,86 @@ pub struct PaymentSwapData {
     pub refund_tx_id: Option<String>,
     pub refund_tx_amount_sat: Option<u64>,
 
+    /// Present only for chain swaps.
+    /// In case of an outgoing chain swap, it's the Bitcoin address which will receive the funds
+    /// In case of an incoming chain swap, it's the Liquid address which will receive the funds
+    pub claim_address: Option<String>,
+
     /// Payment status derived from the swap status
     pub status: PaymentState,
 }
 
+/// The specific details of a payment, depending on its type
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub enum PaymentDetails {
+    Lightning {
+        /// The swap ID, if any swap is associated with this payment
+        swap_id: String,
```

Comment:
Is the documentation wrong here, or the type? As "if any" means this field is nullable.

Reply:
Good catch Erdem, previously it was an `Option` but then I decided to leave it as string since we're sure that if the details exist so will the swap_id (in case of Bitcoin/Lightning details)